app/uk/gov/hmrc/exports/services/NotificationService.scala

```diff
@@ -16,24 +16,22 @@
 
 package uk.gov.hmrc.exports.services
 
-import java.time.format.DateTimeFormatter
-import java.time.{LocalDateTime, ZoneId, ZonedDateTime}
-
 import javax.inject.{Inject, Singleton}
 import play.api.Logger
 import reactivemongo.core.errors.DatabaseException
-import uk.gov.hmrc.exports.models.declaration.notifications.{Notification, NotificationDetails, NotificationError}
-import uk.gov.hmrc.exports.models.declaration.submissions.{Submission, SubmissionStatus}
-import uk.gov.hmrc.exports.models.{NotificationApiRequestHeaders, Pointer, PointerSection, PointerSectionType}
+import uk.gov.hmrc.exports.models.declaration.notifications.Notification
+import uk.gov.hmrc.exports.models.declaration.submissions.Submission
 import uk.gov.hmrc.exports.repositories.{NotificationRepository, SubmissionRepository}
 
 import scala.concurrent.{ExecutionContext, Future}
-import scala.xml.{Node, NodeSeq, XML}
+import scala.xml.XML
 
 @Singleton
-class NotificationService @Inject()(submissionRepository: SubmissionRepository, notificationRepository: NotificationRepository)(
-  implicit executionContext: ExecutionContext
-) {
+class NotificationService @Inject()(
+  submissionRepository: SubmissionRepository,
+  notificationRepository: NotificationRepository,
+  notificationFactory: NotificationFactory
+)(implicit executionContext: ExecutionContext) {
 
   private val logger = Logger(this.getClass)
   private val databaseDuplicateKeyErrorCode = 11000
```

Comment:
I understand this is not related to this PR, but why do we see this value here? A 'Service' should know nothing of a such low-level db-related code.

Reply:
Don't remember exactly why it is service handling the exception and not the repository class. It might be due to reusing one of already existing methods in repository that is more generic and should not handle the exception in all scenarios.